### ISSUE TITLE
Benchmarks for `Any` and fail message propagation

### ISSUE THREAD

**Issue by Kimundi** [rust-lang/rust#10038]
Once #9967 lands, there need to be benchmark written for it and the new fail message propagation system.

**Comment by bltavares** [comment]
Triaging: Would this be a task easy to mentor someone?

**Comment by aturon** [comment]
Closing; not a priority.
